### DISCUSSION TITLE
Pass extra args to shfmt in conform

### DISCUSSION THREAD

**Discussion by kapral18** [LazyVim/LazyVim#1701] [Q&A] [answered]
I want to pass { "-i", "2", "-ci" } to shfmt in conform. But the recommended way with lazy.nvim is to 

```lua
  -- This function is optional, but if you want to customize formatters do it here
  config = function(_, opts)
    local util = require("conform.util")
    util.add_formatter_args(require("conform.formatters.shfmt"), { "-i", "2", "-ci" })
    require("conform").setup(opts)
  end,
```

How can I integrate this into my local [lazyvim](https://github.com/kapral18/config.nvim) starter config without having to copy paste the internals of default lazyvim config into my config?

**Comment by kapral18** [comment]
I see the [update](https://github.com/LazyVim/LazyVim/commit/3eb91c64b5960ccd84bd8e6a3318a1ee79cb85c5). Thanks @folke !

**Reply by folke** [reply, marked as answer]
Yeah, and I also noticed the docs for formatting and linting were not being generated!! :)

Just fixed that. The new docs should be live in a bit.

It already shows the plugins opts, but there's a whole section missing on top.

https://www.lazyvim.org/plugins/formatting

**Reply by folke** [reply]
And new docs are live...

**Comment by dadav** [comment]
What about formatters that need to be configured via env vars?

I did it like this in the past : https://github.com/dadav/dotfiles/blob/master/.config/nvim/lua/plugins/languages/yaml.lua#L44

But since recently, an error pops up to not override config..

**Reply by dadav** [reply]
this error: https://github.com/LazyVim/LazyVim/blob/main/lua/lazyvim/plugins/formatting.lua#L77

**Reply by dadav** [reply]
I know that I could set `--config $external_config`, but maybe there is lazyvim way^^

**Reply by dadav** [reply]
this worked: 
```
      formatters = {
        yaml = vim.tbl_deep_extend("force", require("conform.formatters.yamlfix"), {
          env = {
            YAMLFIX_SEQUENCE_STYLE = "block_style",
            YAMLFIX_WHITELINES = "1",
            YAMLFIX_LINE_LENGTH = "120",
          },
        }),
      },
```

**Reply by dpetka2001** [reply]
Why `require("conform.formatters")`? Aren't you already defining this in the `conform` anyway? Shouldn't a simple `opts.formatters.yamlfix` work instead of the `require`?

**Reply by folke** [reply]
yeah, there's not need to extend, since lazyvim already takes care of that.

Hopefully this will be the default soon for conform, so I can remove LazyVim's deviations.
See https://github.com/stevearc/conform.nvim/pull/140

**Reply by dadav** [reply]
Final version really 2.docx:

```lua
  {
    "stevearc/conform.nvim",
    opts = {
      formatters_by_ft = {
        yaml = { "yamlfix" },
      },
      formatters = {
        yamlfix = {
          env = {
            YAMLFIX_SEQUENCE_STYLE = "block_style",
            YAMLFIX_WHITELINES = "1",
            YAMLFIX_LINE_LENGTH = "120",
          },
        },
      },
    },
  },
```
